daemon/cmd/agenthealth.go

```diff
@@ -65,13 +68,24 @@ func (d *Daemon) startAgentHealthHTTPService(addr string) {
 			}
 			return false
 		}
-
-		statusCode := http.StatusOK
+		currentTs := d.svc.GetCurrentTs()
+		var lastUpdatedTs time.Time
+		var statusCode int
 		sr := d.getStatus(true)
 		if isUnhealthy(&sr) {
+			// If unhealthy, return the timestamp of the last update.
+			lastUpdatedTs = d.svc.GetLastUpdatedTs()
 			statusCode = http.StatusInternalServerError
+		} else {
+			// If healthy, set the lastupdated timestamp to current.
+			statusCode = http.StatusOK
+			lastUpdatedTs = currentTs
 		}
```

Comment:
Please drop the `else` branch an use short variable declaration like the existing code, this is more idiomatic Go IMO. Also I think the comments can be dropped as it is obvious from the code.

```suggestion
		lastUpdatedTs := currentTs
		statusCode := http.StatusOK
		sr := d.getStatus(true)
		if isUnhealthy(&sr) {
			lastUpdatedTs = d.svc.GetLastUpdatedTs()
			statusCode = http.StatusInternalServerError
		}
```

Reply:
Sure I will address your request.

Reply:
Addressed review comments. Made some changes based on Valas reference. It makes sense to split this from the agentHealth so that it is clean.